File: sprint1/Java/lesson7code5.java
public static boolean[] eratosthenesEffective(int n) {
    boolean[] numbers = new boolean[n + 1];
    Arrays.fill(numbers, true);
    numbers[0] = false;
    numbers[1] = false;
    for (int num = 2; num < n; num++) {
        if (numbers[num]) {
            for (int j = num * num; j < n + 1; j += num) {
                numbers[j] = false;
            }
        }
    }
    return numbers;
}

File: sprint1/kotlin/lesson7code5.kt
fun eratosthenesEffective(n: Int): BooleanArray {
    val numbers = BooleanArray(n + 1) { true }
    numbers[0] = false
    numbers[1] = false
    for (num in 2..n) {
        if (numbers[num]) {
            for (j in (num * num)..n step num) {
                numbers[j] = false
            }
        }
    }
    return numbers
}
